hooks/tk-multi-publish2/basic/nuke/publish_files.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import pprint
+import nuke
+import sgtk
+from sgtk.platform import engine
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class NukePublishFilesCustomPlugin(HookBaseClass):
+    """
+    Inherits from NukePublishFilesPlugin
+    """
+    def __init__(self, parent, **kwargs):
+        """
+        Construction
+        """
+        # call base init
+        super(NukePublishFilesCustomPlugin, self).__init__(parent, **kwargs)
+
+    def _sync_frame_range(self):
+        """
+        Checks whether frame range is in sync with shotgun.
+
+        :return: True if yes false otherwise
+        """
+        # get the current engine
+        engine = sgtk.platform.current_engine()
+        # context from engine
+        context = engine.context
+        entity = context.entity
+
+        sg_entity_type = context.entity["type"]
+        sg_filters = [["id", "is", entity["id"]]]
+        fields = ["cut_in", "cut_out"]
+
+        # get the field information from shotgun based on Shot
+        data = engine.shotgun.find_one(sg_entity_type, filters=sg_filters, fields=fields)
+        # compare if the frame range set at root level is same as the shogun cut_in, cut_out
+        root = nuke.Root()
+        if root.firstFrame() != data["cut_in"] or root.lastFrame() != data["cut_out"]:
```

Comment:
You should make sure you handle the case of the data received from Shotgun being invalid...i.e. cut_in/cut_out not being set. Also, are we sure this is the correct validation? Should artists not be publishing with handles? i.e. head_in/tail_out?

Reply:
head_in/tail_out is considered only during mov creation. Here, similar to "Sync Frame Range with Shotgun" option under Shotgun menu, we are checking if the range on the root node of nuke script is same as sg_cut_in and sg_cut_out.

Reply:
In that case, it feels like the more important validation is to check the framerange of the WriteNode's output file sequence rather than the root node, since users have the option to only render out a subset of frames and we should validate that the full framerange is being published.